[docs] Describe workflows and jobs (#518)
someengineering/cloudkeeper: docs/source/manual/automate.rst

---------

Cloudkeeper has a concept of workflows.
A workflow is a set of steps that implements a finite state machine,
where every step performs a specific action.

> The whole workflow can be triggered and is started either by time or by an event.

Reviewer: ```suggestion
Workflows can be triggered on a schedule or by an event.
```

While we want to enable users to create and manipulate workflows on their own, there is currently only one

> hardcoded workflow that ships with cloudkeeper, which can be seen in the following diagram.

Reviewer: ```suggestion
hardcoded workflow that ships with Cloudkeeper, which is illustrated in the following diagram:
```

.. image:: img/default_workflow.svg

This workflow is triggered by time and runs every full hour.
According to this workflow, the following actions are performed every full hour:

- ``collect`` All configured cloud providers are triggered to collect the available cloud resources.
  The collected resources are given to the core and update the internal database.
  After this step is done, all cloud resources should be available to cloudkeeper.

> - ``cleanup_plan`` This step is an entry point to customize the selection of resources, that should be cleaned up.

Reviewer: ```suggestion
- ``cleanup_plan`` This step allows selection of resources that should be cleaned up.
```

>   Cloudkeeper out of the box will select expired resources if ckworker is started with the relevant command

Reviewer: ```suggestion
Out of the box, Cloudkeeper will select expired resources if `ckworker` is started with the relevant command-
```

The following diagram shows, how *ckworker* and *ckmetrics* use the default workflow. *ckworker* waits for the
``collect`` action. Once this is received, it will run all configured cloud providers to collect the available

> cloud resources. It will also listen to the `cleanup_plan` action in the case expired resources should be cleaned up,
> where it would select all resources for cleanup, that are expired according to their expiry definition.

Reviewer: ```suggestion
cloud resources. It will also listen to the `cleanup_plan` action in case expired resources should be cleaned up.
```

    $> query is(resource) and reported.tags.owner==null | tag update owner "John Doe"

> While this is already an improvement, it will only update resources without tags at this moment.

Reviewer: ```suggestion
While this is already an improvement, it will only update resources without tags at the moment.
```
